src/com/linkedin/parseq/TaskDescriptor.java

```diff
@@ -0,0 +1,20 @@
+package com.linkedin.parseq;
+
+/**
+ * An API to provide description for task.
+ *
+ * parseq-lambda-names provide an implementation for this interface using Java SPI. Any changes made to this interface
+ * might require a change to implementation in parseq-lambda-names.
+ *
+ * @author Siddharth Sodhani (ssodhani@linkedin.com)
+ */
+public interface TaskDescriptor {
+
+  /**
+   * Given class which could correspond to generated lambda expressions etc infer appropriate description for it
+   *
+   * @param clazz
+   * @return description for task
+   */
+  String getDescription(Class<?> clazz);
```

Comment:
I am not sure because in that case conversion from Class to String would have to be done by user (parseq main in this case) and String is too generic.